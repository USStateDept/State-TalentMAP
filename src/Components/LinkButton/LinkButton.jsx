import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const LinkButton = ({ children, className, toLink, useDefaultClass, isExternal }) => (
  <span className="link-button-wrapper">
    {
    isExternal ?
      <a
        className={`${useDefaultClass ? 'link-button' : ''} ${className}`}
        type="submit"
        role="button"
        href={toLink}
      >
        {children}
      </a>
      :
      <Link
        className={`${useDefaultClass ? 'link-button' : ''} ${className}`}
        type="submit"
        role="button"
        to={toLink}
      >
        {children}
      </Link>
    }
  </span>
);

LinkButton.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  toLink: PropTypes.string.isRequired,
  useDefaultClass: PropTypes.bool,
  isExternal: PropTypes.bool,
};

LinkButton.defaultProps = {
  className: '',
  useDefaultClass: true,
  isExternal: false,
};

export default LinkButton;
