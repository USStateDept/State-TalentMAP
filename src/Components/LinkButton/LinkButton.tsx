import { Link } from 'react-router-dom';

export type Props = {
  children: React.ReactNode,
  className?: string,
  toLink: string,
  useDefaultClass?: boolean,
  isExternal?: boolean,
};

const LinkButton: React.FC<Props> = props => {
  const { children, className = '', toLink, useDefaultClass = true, isExternal = false } = props;
  return (
    <span className="link-button-wrapper">
      {
        isExternal ?
          <a
            className={`${useDefaultClass ? 'link-button' : ''} ${className}`}
            type="submit"
            role="button"
            href={toLink}
            rel="noopener"
            target="_blank"
          >
            {children}
          </a>
          :
          <Link
            className={`${useDefaultClass ? 'link-button' : ''} ${className}`}
            type="submit"
            role="button"
            to={toLink}
          >
            {children}
          </Link>
      }
    </span>
  );
};

export default LinkButton;
