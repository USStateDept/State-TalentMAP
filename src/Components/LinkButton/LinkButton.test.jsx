import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import LinkButton from './LinkButton';

describe('LinkButtonComponent', () => {
  const child = (<span>Text</span>);
  const toLink = '/test/100';
  it('is defined', () => {
    const wrapper = shallow(
      <LinkButton toLink={toLink}>
        {child}
      </LinkButton>,
    );
    expect(wrapper).toBeDefined();
  });

  it('can take props', () => {
    const wrapper = shallow(
      <LinkButton className="test-class" toLink={toLink}>
        {child}
      </LinkButton>,
    );
    expect(wrapper.instance().props.toLink).toBe(toLink);
    expect(wrapper.find('.test-class')).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <LinkButton toLink={toLink}>
        {child}
      </LinkButton>,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('supports external links', () => {
    const isExternal = true;
    const wrapper = shallow(
      <LinkButton className="test-class" toLink={toLink} isExternal={isExternal}>
        {child}
      </LinkButton>,
    );
    expect(wrapper.instance().props.toLink).toBe(toLink);
    expect(wrapper.find('a.test-class')).toBeDefined();
  });

  it('supports external links with useDefaultClass as false', () => {
    const isExternal = true;
    const useDefaultClass = false;
    const wrapper = shallow(
      <LinkButton toLink={toLink} isExternal={isExternal} useDefaultClass={useDefaultClass}>
        {child}
      </LinkButton>,
    );
    expect(wrapper.instance().props.toLink).toBe(toLink);
    expect(wrapper.find('a')).toBeDefined();
  });

  it('can take props with useDefaultClass as false', () => {
    const useDefaultClass = false;
    const wrapper = shallow(
      <LinkButton toLink={toLink} useDefaultClass={useDefaultClass}>
        {child}
      </LinkButton>,
    );
    expect(wrapper.instance().props.toLink).toBe(toLink);
    expect(wrapper.find('Link')).toBeDefined();
  });
});
