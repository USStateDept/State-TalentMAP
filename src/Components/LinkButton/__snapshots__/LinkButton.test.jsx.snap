// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButtonComponent matches snapshot 1`] = `
<span
  className="link-button-wrapper"
>
  <Link
    className="link-button "
    replace={false}
    role="button"
    to="/test/100"
    type="submit"
  >
    <span>
      Text
    </span>
  </Link>
</span>
`;
